experiment/nursery/main.go

```diff
@@ -142,3 +176,42 @@ func cacheHandler(cache *diskcache.Cache) http.Handler {
 		}
 	})
 }
+
+func monitorDiskAndEvict(cache *diskcache.Cache, interval time.Duration, minBlocksFree, minFilesFree float64) {
+	logger := log.WithFields(log.Fields{
+		"component": "nursery",
+	})
+	dir := cache.DiskRoot()
+	// forever check if usage is past thresholds and evict
+	for range time.Tick(interval) {
+		blocksFree, filesFree := diskutil.GetDiskUsage(dir)
+		// if we are past either threshold, evict until we are not
+		if blocksFree < minBlocksFree || filesFree < minFilesFree {
+			logger.WithFields(log.Fields{
+				"blocks-free": blocksFree,
+				"files-free":  filesFree,
+			}).Warn("Eviction triggered")
+			// get all cache entries and sort by lastaccess
+			// so we can pop entries until we have evicted enough
+			files := cache.GetEntries()
+			sort.Slice(files, func(i, j int) bool {
+				return files[i].LastAccess.After(files[j].LastAccess)
```

Comment:
I think you want `Before` not `After`?

Reply:
Yes. Originally I was popping from the end of the slice. Thanks!